### DISCUSSION TITLE
Tempo Data Source Plugin, Specify a Tag with a Constant Value for Traces to Logs Correlation

### DISCUSSION THREAD

**Discussion by daviddryder** [grafana/grafana#53756] [Observability]
When correlating from a Tempo Trace to a Splunk Log there is no ability to specify the Splunk index name as part of the Tags transferred over to the Splunk Query and thus the query does not work. Adding the index manually to the query results in it working and finding the required logs.

When correlating from a Tempo Trace to a Splunk Log would like the ability to configure a Tag with a constant value in the Tempo datasource Tags to Logs section of the UI configuration.

Currently Tag names can be specified with an Optional Value when the Trace to Logs Data Source is set to Splunk, however the tag and optional value do not appear in the query that is transferred to Splunk. This would allow the Splunk Log Index name to be specified (eg index::prod-logs) when correlating from Traces to Logs, without this the Splunk query does find any results.

Alternatives considered: Using Custom HTTP Headers, to specific the Splunk index, however there does not appear to be a Splunk API Header to support this. Adding the index name as an attribute in the traces, however this requires updating all of the instrumentation.

This feature enhancement request applies to the Tempo Data Source Plugin when using Splunk Logs as the target for Traces to Logs correlation.

**Comment by joe-elliott** [comment]
Thank you for this issue, but this belongs in the Grafana repo. I will move it over there.

**Comment by aocenas** [comment]
Need some help understanding the problem as I have little knowledge of Splunk:
> Currently Tag names can be specified with an Optional Value when the Trace to Logs Data Source is set to Splunk, however the tag and optional value do not appear in the query that is transferred to Splunk. This would allow the Splunk Log Index name to be specified (eg index::prod-logs) when correlating from Traces to Logs, without this the Splunk query does find any results.

I am not sure I understand this part. Does it mean what is needed is something like index::prod-logs to appear in the query? Is the problem then that we don't use :: but = to create key value pairs?

Seems like right now the query looks like this when created tag1=val1 tag2=val2 "traceID" "spanID". Should it be tag1=val1 tag2=val2 "traceID" "spanID" index::some-index?

> however the tag and optional value do not appear in the query that is transferred to Splunk

This part though seems like the general tag mapping functionality does not work for you. Does it mean somebody tried mapping some tag from tempo to index but it still does not appear in the query? Not even as index=val?

**Reply by daviddryder** [reply]
For example in the screenshot below for the (left side) Tempo Traces to Splunk Logs configuration I have added the Tags: service.name, job, ip and index. There is the option to set a default value, so index is set to XYZ

In the Explore (right side) correlation from a Trace to a Log only the Tags: Ip, job and service.name are mapped over to the Splunk query. The index Tag is left out
![Screen Shot 2022-10-26 at 9 45 18 AM](https://user-images.githubusercontent.com/89228383/198118725-8e386453-3d64-4d66-a8a2-90602977d228.png)

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
